src/DIRAC/Resources/IdProvider/IdProvider.py

```diff
@@ -26,3 +33,27 @@ def setParameters(self, parameters):
         """
         self.parameters = parameters
         self.name = parameters.get("ProviderName")
+
+    def getGroupScopes(self, group: str) -> list:
```

Comment:
Well, in theory we can have other types of providers, based on SAML for example

Reply:
Alright, I understand the need to get a flexible and generic architecture, but at the same time, unless I am not aware, we only have OAuth2 use cases for now.
Are we sure that SAML assertions work similarly to OAuth2 tokens? I read that SAML2 "does not help with API authorization" for instance.
Are we sure that this architecture will be enough to support new types of providers?


At least, I think that `getGroupScopes()` and `getScopeGroups()` should be defined in `IdProvider`, empty, and should be overridden in `OAuth2IdProvider` with the current content of the methods (since `scope_to_list()` is bound to OAuth2).
What do you think?

Reply:
I would say that the scope_to_list authlib function is quite a generic helper. If ventually there will be other non-OAuth2 type of providers, this function will be still valid.